content/en/agent/guide/secrets-management.md

```diff
@@ -219,13 +219,178 @@ instances:
     password: decrypted_db_prod_password
 ```
 
-## Troubleshooting secrets
+## Troubleshooting
+
+### Listing detected secrets
+
+The `secret` command in the Agent CLI shows any errors related to your setup—for example, if the rights on the executable are incorrect. It also lists all handles found, as well as where they are located.
+
+On Linux, the command outputs file mode, owner and group for the executable. On Windows, ACL rights are listed.
+
+Example on Linux:
+
+```shell
+$> datadog-agent secret
+=== Checking executable rights ===
+Executable path: /path/to/you/executable
+Check Rights: OK, the executable has the correct rights
+
+Rights Detail:
+file mode: 100700
+Owner username: dd-agent
+Group name: dd-agent
+
+=== Secrets stats ===
+Number of secrets decrypted: 3
+Secrets handle decrypted:
+- api_key: from datadog.yaml
+- db_prod_user: from postgres.yaml
+- db_prod_password: from postgres.yaml
+```
+
+Example on Windows (from an Administrator Powershell):
+```
+PS C:\> & 'C:\Program Files\Datadog\Datadog Agent\embedded\agent.exe' secret
+=== Checking executable rights ===
+Executable path: C:\path\to\you\executable.exe
+Check Rights: OK, the executable has the correct rights
+
+Rights Detail:
+Acl list:
+stdout:
+
+
+Path   : Microsoft.PowerShell.Core\FileSystem::C:\path\to\you\executable.exe
+Owner  : BUILTIN\Administrators
+Group  : WIN-ITODMBAT8RG\None
+Access : NT AUTHORITY\SYSTEM Allow  FullControl
+         BUILTIN\Administrators Allow  FullControl
+         WIN-ITODMBAT8RG\ddagentuser Allow  ReadAndExecute, Synchronize
+Audit  :
+Sddl   : O:BAG:S-1-5-21-2685101404-2783901971-939297808-513D:PAI(A;;FA;;;SY)(A;;FA;;;BA)(A;;0x1200
+         a9;;;S-1-5-21-2685101404-2783901971-939297808-1001)
+
+=== Secrets stats ===
+Number of secrets decrypted: 3
+Secrets handle decrypted:
+- api_key: from datadog.yaml
+- db_prod_user: from sqlserver.yaml
+- db_prod_password: from sqlserver.yaml
+```
+
+### Seeing configurations after secrets were injected
+
+To quickly see how the check's configurations are resolved, you can use the `configcheck` command :
+
+```shell
+sudo -u dd-agent -- datadog-agent configcheck
+
+=== a check ===
+Source: File Configuration Provider
+Instance 1:
+host: <decrypted_host>
+port: <decrypted_port>
+password: <decrypted_password>
+~
+===
+
+=== another check ===
+Source: File Configuration Provider
+Instance 1:
+host: <decrypted_host2>
+port: <decrypted_port2>
+password: <decrypted_password2>
+~
+===
+```
+
+**Note**:  The Agent needs to be restarted to pick up changes on configuration files.
```

Comment:
```suggestion
**Note**: The Agent needs to be restarted to pick up changes on configuration files.
```
might want to link here too - https://docs.datadoghq.com/agent/guide/agent-commands/?tab=agentv6#restart-the-agent